app/lib/models/collection.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+const Api = require(`${paths.lib.models}/api`)
+const CollectionRoutes = require(`${paths.lib.helpers}/collection-routes`)
+const config = require(paths.config)
+
+const Collection = function () {
+
+}
+
+Collection.prototype.buildCollectionRoutes = function () {
+  // new Api(config)
+  return this.getCollections()
+    .then(apiCollections => new CollectionRoutes(apiCollections))
+}
+
+Collection.prototype.getCollections = function () {
+  return Promise.all(
+    config
+      .get('apis')
+      .map(api => new Api(api)
```

Comment:
Personal preference: whenever the content of the arrow function is more complex than a single value, I'd use a normal function body with a `return`, otherwise the indentation is harder to understand. For example:

```js
Collection.prototype.getCollections = function () {
  return Promise.all(
    config
      .get('apis')
      .map(api => {
        return new Api(api)
          .getCollections()
          .then(res => this.getSchemas(api, res.collections))      
      })
  )
}
```